pkg/directory/farms_handlers.go

```diff
@@ -34,7 +40,7 @@ func (s *FarmAPI) registerFarm(r *http.Request) (interface{}, mw.Response) {
 
 	id, err := s.Add(r.Context(), db, info)
 	if err != nil {
-		return nil, mw.Error(err)
+		return nil, mw.GenericDBError(mw.DBError{Cause: err, Message: errFailedToRegisterFarm.Error()})
```

Comment:
can't we have a shortcut method for this ? your input is just the source error, and a message. 
May be something like
```go
mw.GenericDBError(cause, msg) 
```
or even 
```go
mw.GenericDBError(Case, msg, code...) 
```

Reply:
Yes indeed